src/Lti/BasicOutcome/Lti1p3BasicOutcomeProcessor.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+/*
+ *  This program is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU General Public License
+ *  as published by the Free Software Foundation; under version 2
+ *  of the License (non-upgradable).
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ *  Copyright (c) 2021 (original work) Open Assessment Technologies S.A.
+ */
+
+declare(strict_types=1);
+
+namespace OAT\SimpleRoster\Lti\BasicOutcome;
+
+use LogicException;
+use OAT\Library\Lti1p3BasicOutcome\Service\Server\Processor\BasicOutcomeServiceServerProcessorInterface;
+use OAT\Library\Lti1p3BasicOutcome\Service\Server\Processor\Result\BasicOutcomeServiceServerProcessorResult;
+use OAT\Library\Lti1p3Core\Registration\RegistrationInterface;
+use OAT\SimpleRoster\Service\CompleteUserAssignmentService;
+use Psr\Log\LoggerInterface;
+use Throwable;
+
+class Lti1p3BasicOutcomeProcessor implements BasicOutcomeServiceServerProcessorInterface
+{
+    private CompleteUserAssignmentService $completeUserAssignmentService;
+    private LoggerInterface $logger;
+
+    public function __construct(
+        CompleteUserAssignmentService $completeUserAssignmentService,
+        LoggerInterface $logger
+    ) {
+        $this->completeUserAssignmentService = $completeUserAssignmentService;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @inheritDoc
+     *
+     * @SuppressWarnings(PHPMD.UnusedLocalVariable)
+     */
+    public function processReadResult(
+        RegistrationInterface $registration,
+        string $sourcedId
+    ): BasicOutcomeServiceServerProcessorResult {
+        throw new LogicException('Not supported basic outcome result type');
+    }
+
+    /**
+     * @inheritDoc
+     *
+     * @SuppressWarnings(PHPMD.UnusedLocalVariable)
+     */
+    public function processReplaceResult(
+        RegistrationInterface $registration,
```

Comment:
Any Idea why we get the registration? I mean, on which case it would be needed?

Reply:
it's given to you by the LTI framework, to provide more context if needed when you need to deal with basic outcomes operations.

nothing forces you to use it, it's just here for the case you need it :) if you don't use it, just ignore it.

Reply:
It's part of the interface in the lib.

Reply:
I noticed that, my question was more, "why" :)

Reply:
the why was given above :)